plonk-core/src/constraint_system/composer.rs

```diff
@@ -810,6 +810,13 @@ where
             assert_eq!(k, F::zero(), "Check failed at gate {}", i,);
         }
     }
+
+    /// Get value of a variable. Should be safe to `.unwrap()` as long as
+    /// passed `var` was created by this composer before
+    #[inline]
+    pub fn value_of_var(&self, var: Variable) -> Option<F> {
```

Comment:
Not really sure about the fn name. Can't get a better one anyway.. :sweat_smile: 

Also, if unwrapping is safe, let's do so and return `F`directly instead of an `Option<F>`. Using `expect` with a short message about the panic.

Reply:
So this one should be OK?
```rust
    pub fn value_of_var(&self, var: Variable) -> F {
        self.variables.get(&var).cloned().expect("the variable does not exist")
    }
```

Reply:
Also, since you're using `F:Field` you should be able to use `copied` instead of `cloned`.